### PR TITLE
Mega updates

### PR DESCRIPTION
**Pull request by rgbkrk**
Closes #1691, brings RxJS up to date.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -188,38 +188,38 @@
     "chai": "^3.4.1",
     "chai-immutable": "^1.6.0",
     "cross-env": "^5.0.0",
-    "electron": "1.6.6",
-    "electron-builder": "^17.0.0",
+    "electron": "^1.6.8",
```
Comment (lgeiger) [inline package.json]: This should be `"electron": "1.7.1",` otherwise we cannot start nteract because `electron-builder` builds against the wrong ABI.

**Review by lgeiger — COMMENTED** [review]
Looks like we're missing `lodash.debounce` [here](https://github.com/nteract/nteract/blob/master/packages/editor/package.json#L29).

**Comment by rgbkrk** [comment]
> Looks like we're missing lodash.debounce

I see how I made that fail now. Taking out a dependency that had it (react codemirror) that we were no longer using since I migrated it. 🤦‍♂️

**Comment by lgeiger** [comment]
I guess we need to update `yarn.lock` to use the latest version of `node-abi` to support Electron 1.7+.

**Comment by rgbkrk** [comment]
Running `yarn upgrade` now.

**Comment by rgbkrk** [comment]
More for later:

```
lerna WARN EHOIST_ROOT_VERSION The repository root depends on codemirror@^5.25.2, which differs from the more common codemirror@~5.24.0.
lerna WARN EHOIST_PKG_VERSION "@nteract/editor" package depends on codemirror@~5.24.0, which differs from the hoisted codemirror@^5.25.2.
lerna WARN EHOIST_PKG_VERSION "@nteract/notebook-preview-demo" package depends on codemirror@~5.24.0, which differs from the hoisted codemirror@^5.25.2.
lerna WARN EHOIST_PKG_VERSION "@nteract/notebook-preview" package depends on codemirror@~5.24.0, which differs from the hoisted codemirror@^5.25.2.
lerna WARN EHOIST_ROOT_VERSION The repository root depends on prop-types@^15.5.10, which differs from the more common prop-types@^15.5.8.
lerna WARN EHOIST_PKG_VERSION "@nteract/editor" package depends on prop-types@^15.5.8, which differs from the hoisted prop-types@^15.5.10.
lerna info Installing external dependencies
```

**Comment by rgbkrk** [comment]
Travis passes yet appveyor fails.

```

C:\projects\nteract\packages\notebook-preview\src\markdown-cell.js
  20:33  error  `tabIndex` should only be declared on interactive elements  jsx-a11y/no-noninteractive-tabindex
C:\projects\nteract\src\notebook\components\cell\markdown-cell.js
  140:11  error  `tabIndex` should only be declared on interactive elements  jsx-a11y/no-noninteractive-tabindex
C:\projects\nteract\src\notebook\views\toolbar.js
  100:23  error  Non-interactive elements should not be assigned interactive roles  jsx-a11y/no-noninteractive-element-to-interactive-role
  108:23  error  Non-interactive elements should not be assigned interactive roles  jsx-a11y/no-noninteractive-element-to-interactive-role
  116:23  error  Non-interactive elements should not be assigned interactive roles  jsx-a11y/no-noninteractive-element-to-interactive-role
  124:23  error  Non-interactive elements should not be assigned interactive roles  jsx-a11y/no-noninteractive-element-to-interactive-role
  135:19  error  Non-interactive elements should not be assigned interactive roles  jsx-a11y/no-noninteractive-element-to-interactive-role
✖ 7 problems (7 errors, 0 warnings)
```

I'll see if I can either take care of that or just make appveyor do less.

**Comment by rgbkrk** [comment]
```
 npm ERR! Error: EPERM: operation not permitted, rename 'C:\Users\appveyor\AppData\Roaming\npm-cache\codemirror\5.24.2\package.tgz.2720672997' -> 'C:\Users\appveyor\AppData\Roaming\npm-cache\codemirror\5.24.2\package.tgz'
lerna ERR! execute npm ERR!     at Error (native)
lerna ERR! execute npm ERR!  { Error: EPERM: operation not permitted, rename 'C:\Users\appveyor\AppData\Roaming\npm-cache\codemirror\5.24.2\package.tgz.2720672997' -> 'C:\Users\appveyor\AppData\Roaming\npm-cache\codemirror\5.24.2\package.tgz'
lerna ERR! execute npm ERR!     at Error (native)
lerna ERR! execute npm ERR!   errno: -4048,
lerna ERR! execute npm ERR!   code: 'EPERM',
lerna ERR! execute npm ERR!   syscall: 'rename',
lerna ERR! execute
```

**Comment by rgbkrk** [comment]
All set now, cache was stale on Appveyor.

**Review by lgeiger — APPROVED** [review]
Thanks 👍 🚀

**Pull request merged**